src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs

```diff
@@ -0,0 +1,36 @@
+#if (NativeAot)
+using System.Text.Json.Serialization;
+#endif
+using Company.ApiApplication1;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Logging.AddConsole();
+
+#if (NativeAot)
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+});
+
+#endif
+var app = builder.Build();
+
+var sampleAlbums = AlbumGenerator.GenerateAlbums().ToArray();
+
+var api = app.MapGroup("/albums");
+api.MapGet("/", () => sampleAlbums);
+api.MapGet("/{id}", (int id) =>
+    sampleAlbums.FirstOrDefault(a => a.Id == id) is { } album
+        ? Results.Ok(album)
+        : Results.NotFound($"Album with id {id} not found"));
+
+app.Run();
+
+#if (NativeAot)
+[JsonSerializable(typeof(string))]
```

Comment:
I think we can leave the primitive types off. The source generator takes care of the transitivity. It's kind of weird to call out `string`, but not have all the other types in Album.

Technically, we'd only need `Album[]`.

Reply:
Yeah this is one of those areas where what's correct/better could be debated. I'm going to remove the result that relies on string and if folks think it's better to just specify the array type here I'm happy to change it to that.